lib/api/pokedex_api_spec/api_service.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:pokedex/api/pokedex_api_spec/api_client.dart';
+import 'package:pokedex/api/pokedex_api_spec/handler/pokemon_handler.dart';
+
+class ApiService {
+  static final ApiService _apiService = ApiService._internal();
+
+  factory ApiService() => _apiService;
+
+  ApiService._internal();
+
+  late final ApiClient _apiClient = ApiClient();
+
+  late final PokemonApi pokemonApi = PokemonApi(_apiClient);
```

Comment:
no. not like this. it sucks.

Reply:
this should be just like this. 

```dart
late PokemonApi _pokemonApi;
```

then above you need to create a getter.

```dart
PokemonApi get pokemonApi => _pokemonApi ??= PokemonApi(_apiClient);
```

what `??=` does is it checks `_pokemonApi` if not null it will return the instance of `_pokemonApi` **BUT** if it's null it will assign  `PokemonApi(_apiClient)` so the next time you call the getter `_pokemonApi` already has value.

Reply:
Is this to avoid creating multiple instance?

Reply:
should I make _pokemonApi nullable?

Reply:
something like this late `late PokemonApi? _pokemonApi;`

Reply:
```dart
class ApiService {
  ...

  ...

  ...
  /// This is the one to be accessed publicly.
  PokemonApi get pokemonApi => _pokemonApi ??= PokemonApi(_apiClient);

  late final ApiClient _apiClient = ApiClient();
  late PokemonApi _pokemonApi;
}
```

you should be having something like this @yohancai

Reply:
Does singleton will also solve that issue?